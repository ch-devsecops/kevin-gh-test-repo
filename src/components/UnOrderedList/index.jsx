import React from 'react';
import { compiler } from 'markdown-to-jsx';
import { UnorderedList, UnorderedListItem, Box, Markdown, H4 } from '@honda-canada/design-system-react';
import { wrapJSSFields } from '../../utils/wrapJSSFields';
import { colourTokenForParam, mapGTMCategory } from '../../utils/sitecoreFields';
import { stripMarkdownHeading, getTitleComponent } from '../../utils/markdown';

const UnOrderedList = ({ fields, params = {}, rendering = {} }) => {
  if (!fields) return null;

  const { title, anchorId, gtmTitle, gtmCategory } = wrapJSSFields(fields);
  const items = fields.items?.map(item => wrapJSSFields(item.fields));

  const columnsOnDesktopInteger = params?.columnsOnDesktop === 'Two' ? 2 : 1;
  const marginTop = params?.hasNegativeMargin && params?.negativeMargin ? `-${params?.negativeMargin}px` : 'zero';
  const titleAlignment = params?.contentAlignment === 'Center' ? params?.contentAlignment : 'Left';
  const bulletStyle = params?.bulletStyle === 'Bullet-Points' ? 'bullet' : 'checkmark';

  // magic numbers for the maximum width of things
  const componentWidth = '824px';
  const listWidth = '612px';

  const pb = rendering.params?.pb || ['xl', 'xxl'];
  const px = rendering.params?.px || ['default', 'zero'];
  const TitleComponent = getTitleComponent(title?.value, H4);

  return (
    <Box
      data-gtm-title={gtmTitle?.value}
      data-gtm-category={mapGTMCategory(gtmCategory)}
      anchorId={anchorId?.value}
      width={['100%', 'auto']}
      mt={marginTop}
      pb={pb}
      px={px}
      bg={colourTokenForParam[params?.bgColour?.toLowerCase()]}
      alignItems="center"
    >
      <Box maxWidth={['initial', componentWidth]} mx="auto">
        {title?.value && (
          <Box mb={['xs', 'm']}>
            <TitleComponent textAlign={titleAlignment}>{compiler(stripMarkdownHeading(title?.value))}</TitleComponent>
          </Box>
        )}
        <Box maxWidth={listWidth} mx="auto">
          <UnorderedList bulletStyle={bulletStyle} columns={[1, columnsOnDesktopInteger]}>
            {items.map((item, i) => (
              <UnorderedListItem key={i.toString()}>
                <Markdown>{item?.text?.value}</Markdown>
              </UnorderedListItem>
            ))}
          </UnorderedList>
        </Box>
      </Box>
    </Box>
  );
};

export default UnOrderedList;
