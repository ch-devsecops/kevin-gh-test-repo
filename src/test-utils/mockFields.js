/* eslint-disable max-len */
/* eslint-disable max-classes-per-file */

// FIXME: jsdoc popup not working on test files
/** Class representing a mock data. */
class MockData {
  /**
   * Add a field.
   * @param {string} key - key of the field obj: i.e. 'title'
   * @param {Object} value - value of the field obj: i.e. { value: 'some text' }.
   */
  add(key, value) {
    this[key] = value;
  }
}

/**
 * Class representing mock fields object.
 * @extends MockData
 *
 * available method: add()
 *
 * default fields: [anchorId, title, bodyText, image, ctaLink, ctaType, ctaIcon, gtmCategory, gtmTitle, contentTitle, subtitle]
 *
 * note: we can add more commonly used fields in the class declaration if needed
 */
class MockFields extends MockData {
  constructor() {
    super();

    this.title = { value: 'Title' };
    this.bodyText = { value: 'BodyText' };
    this.image = {
      value: { src: 'mock.png', alt: 'mock alt' },
    };
    this.ctaLink = {
      value: {
        href: 'www.source.ca',
        title: 'cta aria label',
        text: 'CTA Text',
      },
    };
    this.ctaType = {
      value: 'Tertiary',
    };
    this.ctaIcon = {
      fields: {
        value: {
          value: 'arrowRight',
        },
      },
    };
    this.gtmCategory = {
      fields: {
        value: {
          value: 'Others',
        },
      },
    };
    this.gtmTitle = {
      value: 'gtm title',
    };
    this.contentTitle = {
      value: 'Content Title',
    };
    this.subtitle = {
      value: 'Content Subtitle',
    };
    this.seoH1 = {
      value: '',
    };
    this.anchorId = {
      value: 'anchor-id',
    };
    this.ctaLabel = {
      value: 'CTA Label',
    };
  }
}

/**
 * Class representing a mock rendering object
 * @extends MockData
 *
 * available method: add()
 */
class MockRendering extends MockData {
  /**
   * Create a mock rendering object
   * @param {string} componentName - adds componentName
   */
  constructor(componentName) {
    super();

    this.componentName = componentName;
  }
}

export { MockFields, MockRendering };
