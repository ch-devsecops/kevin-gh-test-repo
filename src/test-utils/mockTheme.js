/**
 * Mock styled-components theme, if you want to test how your component
 * implements theme styles. Import as-is, or spread in any
 * additional values you need.
 *
 * Example usage:

  it('should apply the bgColor param', () => {
    const { colors } = MockTheme;
    const { container } = render(
      <MyComponent
        fields={mockFields}
        params={{ bgColor: 'Gray' }}
      />,
    );
    const wrapper = container.querySelector('section');

    expect(wrapper).toHaveStyleRule('background-color', colors.grey[5]);
  });
 */

import { HONDA_THEME_NAME } from '../utils/constants';

const mockTheme = {
  letterSpacings: {
    sm: '0.5px',
    md: '1px',
  },
  lineHeights: {
    sm: '22px',
    md: '26px',
  },
  colors: {
    white: '#fff',
    transparent: 'transparent',
    grey: ['#707070', '#C7C7C7', '#D5D5D5', '#EAEAEA', '#F8F8F8', '#F5F5F5'],
    black: '#2C2C2C',
    red: '#DA2C2E',
    redShades: ['rgba(200, 16, 46, 0.1)'],
    aodaRed: '#D92629',
    lightRed: '#EC9597',
    darkRed: '#A21C1D',
    blue: '#334C96',
    darkBlue: '#002D62',
    lightBlue: '#00447826',
    yellow: ['#F9CA52', '#EDAB00', '#8D763F'],
    fadedBlue: '#ADB7D5',
    primary: '#DA2C2E',
    primaryAoda: '#D92629',
    primaryDark: '#A21C1D',
    primaryLight: '#EC9597',
    secondary: '#2C2C2C',
    secondaryLight: '#C7C7C7',
    tertiary: '#002D62',
    tertiaryLight: '#334C96',
    tertiaryDisabled: '#ADB7D5',
    danger: '#DA2C2E',
    warning: '#2C2C2C',
    warningLight: '#EAEAEA',
    success: '#334C96',
    successLight: '#d9e0e7',
    disabled: '#C7C7C7',
    typographyDefault: '#2C2C2C',
  },
  space: [0, 4, 8, 12, 16, 24, 32, 48, 64, 96, 128, 144, 192, 256],
  borders: [0, '1px solid', '2px solid', '3px solid'],
  breakpoints: ['768px', '992px', '1200px'],
  fontWeights: {
    regular: 'normal',
    bold: 700,
  },
  fontSizes: {
    sm: '12px',
  },
  shadows: {
    smooth: {
      md: '0.1px 0.3px 0.3px rgba(0, 0, 0, 0.02),0.3px 0.7px 0.9px rgba(0, 0, 0, 0.028),0.7px 1.4px 1.8px rgba(0, 0, 0, 0.035),1.5px 2.9px 3.7px rgba(0, 0, 0, 0.047),4px 8px 10px rgba(0, 0, 0, 0.07)',
    },
    box: {
      subNav: '0px 2px 4px rgba(0, 0, 0, 0.1)',
    },
  },
  zIndices: {
    accordionItemTitle: 200,
    modal: 13001,
    sectionNav: 13000,
  },
  transitionDuration: {
    t1: '1s',
  },
  transitionTimingFunction: {
    default: 'ease-in-out',
    out: 'ease-out',
    in: 'ease-in',
  },
  name: HONDA_THEME_NAME,
  fonts: {
    default: 'Helvetica',
    bold: 'Cooper Black',
    heading: 'Roboto',
  },
  header: {
    desktop: {},
    mobile: {},
  },
  icons: {},
  logIconWarnings: false,
};

export default mockTheme;
