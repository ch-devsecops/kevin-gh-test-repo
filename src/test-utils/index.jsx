import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { BrowserRouter } from 'react-router-dom';
import { OptionalVideoModalContextProvider } from '@honda-canada/design-system-react';
import UserLocationProvider from '@honda-canada/user-location';
import { configureAxe } from 'jest-axe';
import ModelExplorationContextProvider from '../components/ModelExplorationContext';
import ModelFiltersContextProvider from '../components/ModelFiltersContext';
import { MockFields, MockRendering } from './mockFields';
import MockTheme from './mockTheme';
import 'intersection-observer';
import { ACURA_THEME_NAME } from '../utils/constants';

const generateJssWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ACURA_THEME_NAME },
          ...context?.sitecoreContext,
        }}
        componentFactory={{}}
      >
        <BrowserRouter>
          <UserLocationProvider>
            <ThemeProvider theme={context?.theme || MockTheme}>
              <OptionalVideoModalContextProvider>
                <ModelExplorationContextProvider>
                  <ModelFiltersContextProvider>{children}</ModelFiltersContextProvider>
                </ModelExplorationContextProvider>
              </OptionalVideoModalContextProvider>
            </ThemeProvider>
          </UserLocationProvider>
        </BrowserRouter>
      </SitecoreContext>
    );
  };
/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in an typical JSS app's context providers.
 * Note that you can include a `context` object to set or override values
 * for SitecoreContext and ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.context] - default values for SitecoreContext, ThemeProvider
 * @param {string} [options.context.sitecoreContext] - value for SitecoreContext
 * @param {string} [options.context.theme] - value for ThemeProvider
 * @example
 * const { getByText } = jssRender(
 *  <MyComponent />, { context: { sitecoreContext: { language: 'fr' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const jssRender = (component, options = {}) => {
  const { context } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.context;

  return render(component, {
    wrapper: generateJssWrapper(context),
    ...options,
  });
};

const generateWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <BrowserRouter>
        <ThemeProvider theme={context?.theme || MockTheme}>
          <OptionalVideoModalContextProvider>{children}</OptionalVideoModalContextProvider>
        </ThemeProvider>
      </BrowserRouter>
    );
  };

/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in typical design-system context providers.
 * Note that you can include a `theme` object to set or override values
 * for ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.theme] - default value for ThemeProvider
 * @example
 * const { getByText } = render(
 *  <MyComponent />, { theme: { space: { wrapperGap: '15px' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const customRender = (component, options = {}) => {
  const { theme } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.theme;

  return render(component, {
    wrapper: generateWrapper(theme),
    ...options,
  });
};

// disable accessibility checks for errors that are usually trigged by content mistakes
const axe = configureAxe({
  rules: {
    'duplicate-id': { enabled: false },
    'heading-order': { enabled: false },
  },
});

export * from 'jest-axe';
export { axe };
export * from '@testing-library/react';
export { customRender as render, jssRender };
export { MockFields, MockRendering, MockTheme };
