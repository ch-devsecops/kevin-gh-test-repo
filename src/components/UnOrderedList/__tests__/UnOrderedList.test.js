import React from 'react';
import { jssRender, cleanup, MockTheme } from 'test-utils';
import UnOrderedList from '../index';

const mockData = require('./mockData.test.json');

afterEach(cleanup);

describe('<UnOrderedList>', () => {
  const { colors } = MockTheme;

  test('should render correctly', () => {
    const { container } = jssRender(<UnOrderedList fields={mockData.fields} />);

    expect(container).toMatchSnapshot();
    expect(container.children[0]).toBeTruthy();
  });

  test('should render as center aligned', () => {
    const { container } = jssRender(<UnOrderedList fields={mockData.fields} params={{ contentAlignment: 'Center' }} />);
    const title = container.querySelector('h4');

    expect(title).toHaveStyle({ 'text-align': 'Center' });
  });

  test('should render blue background colour', () => {
    const { container } = jssRender(<UnOrderedList fields={mockData.fields} params={{ bgColour: 'Blue' }} />);
    const wrapper = container.querySelectorAll('div')[0];

    expect(wrapper).toHaveStyle({ 'background-color': colors.blue });
  });

  test('should render as null without fields', () => {
    const { container } = jssRender(<UnOrderedList />);

    expect(container.firstChild).toBeNull();
  });
});
