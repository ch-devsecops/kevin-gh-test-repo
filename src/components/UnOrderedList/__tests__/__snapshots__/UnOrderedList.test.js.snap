// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnOrderedList> should render correctly 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 ghzkol"
    data-gtm-category="Others"
    data-gtm-title=""
    width="100%,auto"
  >
    <div
      class="Box-sc-1bsf784-0 jHNFVP"
    >
      <div
        class="Box-sc-1bsf784-0 gKqomd"
      >
        <h4
          class="sc-1k44dfv-0 VXyJj"
        >
          <span>
            List with bullet points
          </span>
        </h4>
      </div>
      <div
        class="Box-sc-1bsf784-0 dgZhPl"
      >
        <ul
          class="sc-1k44dfv-0 eGrUBH"
        >
          <li
            class="sc-1k44dfv-0 sc-1k44dfv-0 UnorderedListItem-sc-1h5wvnk-0 hJLbMT"
            color="typographyDefault"
            font-family="default"
            font-weight="regular"
          >
            <div
              class="sc-1k44dfv-0 klHKZs"
            >
              

              <p>
                The first item ipsum dolor amet, consectetur adipiscing elit, sed do ut labore et dolore magna. The first item ipsum dolor amet, consectetur adipiscing elit, sed do ut labore et dolore magna. The first item ipsum dolor amet, consectetur adipiscing elit, sed do ut labore et dolore magna.
              </p>
            </div>
          </li>
          <li
            class="sc-1k44dfv-0 sc-1k44dfv-0 UnorderedListItem-sc-1h5wvnk-0 hJLbMT"
            color="typographyDefault"
            font-family="default"
            font-weight="regular"
          >
            <div
              class="sc-1k44dfv-0 klHKZs"
            >
              

              <p>
                List Item 2
              </p>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
